src/bundler/bundle_v2.zig

```diff
@@ -3840,7 +3841,15 @@ const LinkerContext = struct {
                 const pathname = Fs.PathName.init(this.graph.entry_points.items(.output_path)[chunk.entry_point.entry_point_id].slice());
                 chunk.template.placeholder.name = pathname.base;
                 chunk.template.placeholder.ext = "js";
-                chunk.template.placeholder.dir = pathname.dir;
+
+                var dir = std.fs.cwd().openDir(pathname.dir, .{}) catch |err| {
```

Comment:
1) Do we need to do this? Opening directories is expensive. If the original file was a symlink, it would produce unexpected results - it would become the realpath instead of the input path. Instead, IMO, we should join the input path with the root path and choose based on that.
2) If we do need to do this, don't use `openDir` because it asks for more permissions than necessary on Linux. Instead, we should add a version of `openFileForPath` that adds the `O.DIRECTORY` flag.

 https://github.com/oven-sh/bun/blob/a008e49a40565c1481b4216bfa9d60016de8b568/src/bun.zig#L1501

Reply:
re: symlink, another option is to pass the flag to not follow symlink path - so it would open the symlink instead of the real path

additionally, we should have a hash table here that checks for duplicate dirname